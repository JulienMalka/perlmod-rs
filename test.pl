#!/usr/bin/env perl

use v5.28.0;

use lib '.';
use RSPM::Bless;
use RSPM::Foo142;

my $v = RSPM::Bless->new("Hello");
$v->something();
my ($a, $b, $c) = $v->multi_return();
say "Got ($a, $b, $c)";
my @ret = $v->multi_return();
say "Got: ".scalar(@ret)." values: @ret";

$v->another(54);

my $param = { a => 1 };
my $s = "Hello You";
print "These should be called with a valid substr:\n";
RSPM::Foo142::test(substr($s, 3, 3));
RSPM::Foo142::teststr(substr($s, 3, 3));
print "Parameter exists: " . (exists($param->{x}) ? "YES" : "NO") . "\n";
RSPM::Foo142::test($param->{x});
print "Was auto-vivified: " . (exists($param->{x}) ? "YES" : "NO") . "\n";
RSPM::Foo142::teststr($param->{x});
