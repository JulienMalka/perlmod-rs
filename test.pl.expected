Called something on Bless { "Hello" }!
Called 'another(54)' on Bless { "Hello" }!
test called with Some("lo ")
teststr called with Some("lo ")
test called with None
teststr called with None
Got (17, 32, )
Got: 2 values: 17 32
These should be called with a valid substr:
Parameter exists: NO
Was auto-vivified: NO
